File: One Task/Task 1/c programming/55.c
//  Check for a Magic Square

#include <stdio.h>

int isMagicSquare(int mat[3][3])
{
    int sumd1 = 0, sumd2 = 0;

    for (int i = 0, j = 2; i < 3; i++, j--) {
        sumd1 += mat[i][i];
        sumd2 += mat[i][j];
    }

    if(sumd1 != sumd2)
        return 0;

    for (int i = 0; i < 3; i++) {
        int rowSum = 0, colSum = 0;

        for (int j = 0; j < 3; j++) {
            rowSum += mat[i][j];
            colSum += mat[j][i];
        }

        if (rowSum != colSum || colSum != sumd1)
            return 0;
    }

    return 1;
}

int main()
{
	int mat[3][3] = {{ 2, 7, 6 },
					{ 9, 5, 1 },
					{ 4, 3, 8 }};
	
	if (isMagicSquare(mat) == 1)
		printf("Magic Square");
	else
		printf("Not a magic Square");
	
	return 0;
}

File: Without Node Module/Xampp/htdocs/core/week-1/c-programs/programs/55.php
<?php
    function isMagicSquare($matrix) {
        $n = count($matrix);
        $expectedSum = $n * ($n * $n + 1) / 2;

        for ($i = 0; $i < $n; $i++) {
            $rowSum = $colSum = 0;
            for ($j = 0; $j < $n; $j++) {
                $rowSum += $matrix[$i][$j];
                $colSum += $matrix[$j][$i];
            }

            if ($rowSum != $expectedSum || $colSum != $expectedSum) {
                return false;
            }
        }

        $primaryDiagonalSum = $secondaryDiagonalSum = 0;
        for ($i = 0; $i < $n; $i++) {
            $primaryDiagonalSum += $matrix[$i][$i];
            $secondaryDiagonalSum += $matrix[$i][$n - $i - 1];
        }

        if ($primaryDiagonalSum != $expectedSum || $secondaryDiagonalSum != $expectedSum) {
            return false;
        }

        echo "Primary Diagonal Sum: $primaryDiagonalSum<br>";
        echo "Secondary Diagonal Sum: $secondaryDiagonalSum<br>";

        return true;
    }

    function printMatrix($matrix) {
        foreach ($matrix as $row) {
            echo implode(", ", $row) . "<br>";
        }
    }

    $matrix = [
        [2, 7, 6],
        [9, 5, 1],
        [4, 3, 8]
    ];

    printMatrix($matrix);

    if (isMagicSquare($matrix)) {
        echo "It's a magic square!";
    } else {
        echo "It's not a magic square.";
    }
?>
